src/main/kotlin/org/rust/cargo/runconfig/legacy/RsAsyncRunner.kt

```diff
@@ -144,10 +147,11 @@ abstract class RsAsyncRunner(
         isTestBuild: Boolean
     ): Promise<Binary?> {
         val promise = AsyncPromise<Binary?>()
-        val cargo = state.cargo()
+        val toolchain = state.toolchain
+        val cargo = toolchain.cargo()
```

Comment:
Why don't you use `org.rust.cargo.runconfig.CargoRunStateBase#cargo` anymore here? Note, under the hood, `org.rust.cargo.runconfig.CargoRunStateBase#cargo` calls `toolchain.cargoOrWrapper(workingDirectory)`, i.e. provide additional support for xargo